### PR TITLE
parser: fix limit lost in SetOprStmt

### LINKED ISSUE #49874: Nested union with LIMIT handles incorrectly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE a(PK int);
CREATE TABLE b(PK int);

INSERT INTO a VALUES (1), (2);
INSERT INTO b VALUES (3), (4);

select * from a where PK = 0
union all
  (
    select * from b where PK = 0
    union all
    (select * from b where PK != 0)
    order by pk
    limit 1
  )
order by pk
limit 2
```

### 2. What did you expect to see? (Required)

MySQL 8.0:

```
+------+
| PK   |
+------+
|    3 |
+------+
1 row in set (0.00 sec)
```

### 3. What did you see instead (Required)

```
mysql> select * from a where PK = 0
    -> union all
    ->   (
    ->     select * from b where PK = 0
    ->     union all
    ->     (select * from b where PK != 0)
    ->     order by pk
    ->     limit 1
    ->   )
    -> order by pk
    -> limit 2;
+------+
| PK   |
+------+
|    3 |
|    4 |
+------+
2 rows in set (0.00 sec)
```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #49874

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/parser.go [hunk 1]
```diff
@@ -19931,15 +19931,19 @@ yynewstate:
 			}
 			var setOprList2 []ast.Node
 			var with2 *ast.WithClause
+			var limit2 *ast.Limit
```
Comment (hawkingrei) [inline pkg/parser/parser.go:19934]: Can you add test cases for it ？

Comment (Defined2014) [reply pkg/parser/parser.go:19934]: Yes, I'm doing it.

Comment (Defined2014) [reply pkg/parser/parser.go:19934]: Done

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**